tools/src/main/scala/org/apache/spark/sql/rapids/tool/AppFilterImpl.scala

```diff
@@ -71,21 +71,38 @@ class AppFilterImpl(
 
     val filterAppName = appArgs.applicationName.getOrElse("")
     if (appArgs.applicationName.isSupplied && filterAppName.nonEmpty) {
-      val filtered = apps.filter { app =>
-        val appNameOpt = app.appInfo.map(_.appName)
-        if (appNameOpt.isDefined) {
-          appNameOpt.get.equals(filterAppName)
-        } else {
-          // in complete log file
-          false
-        }
+      val checkNegation = filterAppName(0)
+      val filtered = if (checkNegation.equals('~')) {
+        apps.filterNot(app => filterAppsNegate(app, filterAppName))
+      } else {
+        apps.filter(app => filterApps(app, filterAppName))
       }
       filtered.map(_.eventlog).toSeq
     } else {
       apps.map(x => x.eventlog).toSeq
     }
   }
 
+  def filterApps(app: AppFilterReturnParameters, filterAppName: String): Boolean = {
+    val appNameOpt = app.appInfo.map(_.appName)
+    if (appNameOpt.isDefined) {
+      appNameOpt.get.equals(filterAppName)
```

Comment:
@tgravescs - I am doing exact match here and for negation method below. 
Should it be a match even if it's a substring? I mean should I change equals to contains ?

Reply:
we should change both to contains.. so that its not exact equals but could be substring, that is what requirements say